cms/djangoapps/contentstore/views/preview.py

```diff
@@ -213,10 +214,25 @@ def _preview_module_system(request, descriptor, field_data):
             "i18n": ModuleI18nService,
             "settings": SettingsService(),
             "user": DjangoXBlockUserService(request.user),
+            "partitions": PreviewPartitionService(user=request.user, course_id=course_id)
```

Comment:
@nasthagiri I needed to add a Studio version of the service or else viewing the components in Studio failed (due to mixin further down).